syntax = "proto3";

package f1;

service Monitoramento {
  // Mudamos aqui: agora aceita uma lista de dados
  rpc EnviarLotePneus (ListaDadosCarro) returns (Resposta) {}
}

message ListaDadosCarro {
  repeated DadosCarro dados = 1; // "repeated" significa Lista/Array em Protobuf
}

message Pneu {
  float temperatura = 1;
  float desgaste = 2;
  float pressao = 3;
}

message DadosCarro {
  string carro_id = 1;
  string sensor_id = 2;
  float velocidade = 3;
  int32 volta = 4;
  string timestamp = 5;
  Pneu pneu_fl = 6;
  Pneu pneu_fr = 7;
  Pneu pneu_rl = 8;
  Pneu pneu_rr = 9;
}

message Resposta {
  string mensagem = 1;
  bool sucesso = 2;
}